### PR TITLE
Actions

### PR DESCRIPTION
**Pull request by jburel**
Replace travis by GHA
use tox file

### CODE REVIEW

File: .github/workflows/tox.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      fail-fast: false
+      matrix:
+        python-version:
+          - '3.6'
+          - '3.7'
+          - '3.8'
+          - '3.9'
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
```
Comment (manics) [inline .github/workflows/tox.yml:20]: Can you add a pre-commit step here?

File: tox.ini [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    pytest-rerunfailures
+    pytest-xdist
+    restructuredtext-lint
+    https://github.com/ome/zeroc-ice-py-github-ci/releases/download/0.1.0/zeroc_ice-3.6.5-cp36-cp36m-linux_x86_64.whl; platform_system=="Linux" and python_version=="3.6"
+    https://github.com/ome/zeroc-ice-py-github-ci/releases/download/0.1.0/zeroc_ice-3.6.5-cp37-cp37m-linux_x86_64.whl; platform_system=="Linux" and python_version=="3.7"
+    https://github.com/ome/zeroc-ice-py-github-ci/releases/download/0.1.0/zeroc_ice-3.6.5-cp38-cp38-linux_x86_64.whl; platform_system=="Linux" and python_version=="3.8"
+    https://github.com/ome/zeroc-ice-py-github-ci/releases/download/0.1.0/zeroc_ice-3.6.5-cp39-cp39-linux_x86_64.whl; platform_system=="Linux" and python_version=="3.9"
+commands =
+    rst-lint README.rst
+    flake8 .
```
Comment (manics) [inline tox.ini:22]: pre-commit is used instead of flake8

**Comment by jburel** [comment]
Added pre-commit in tox with black and flake8

**Comment by manics** [comment]
One comment about a line that's not clear, but otherwise looks fine to me

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    https://github.com/ome/zeroc-ice-py-github-ci/releases/download/0.1.0/zeroc_ice-3.6.5-cp36-cp36m-linux_x86_64.whl; platform_system=="Linux" and python_version=="3.6"
+    https://github.com/ome/zeroc-ice-py-github-ci/releases/download/0.1.0/zeroc_ice-3.6.5-cp37-cp37m-linux_x86_64.whl; platform_system=="Linux" and python_version=="3.7"
+    https://github.com/ome/zeroc-ice-py-github-ci/releases/download/0.1.0/zeroc_ice-3.6.5-cp38-cp38-linux_x86_64.whl; platform_system=="Linux" and python_version=="3.8"
+    https://github.com/ome/zeroc-ice-py-github-ci/releases/download/0.1.0/zeroc_ice-3.6.5-cp39-cp39-linux_x86_64.whl; platform_system=="Linux" and python_version=="3.9"
+commands =
+    pre-commit run --all-files
+    rst-lint README.rst
+    python setup.py sdist install --record files.txt
+    pytest {posargs:-n4 -rf tests -s}
+    sh -c 'cat files.txt | xargs rm -rf'
```
[the remark this answers is not in the document]
Comment (jburel) [reply tox.ini]: removed

**Review by manics — APPROVED** [review]

**Pull request merged**